src/Core/Checkout/Test/Cart/Tax/TaxDetectorTest.php

```diff
@@ -177,7 +177,50 @@ public function testIsNotNetDeliveryWithCompanyFreeTaxAndWrongVatIdPattern(): vo
         $detector = $this->getContainer()->get(TaxDetector::class);
         static::assertFalse($detector->isNetDelivery($context));
     }
+    public function testIsNotNetDeliveryWithCompanyFreeTaxAndNullVatId(): void
```

Comment:
You don't need to mock that much here, as you can create partial entities without the need of mocking.
And if you mock everything there should be no need to hid the DB
i would propose something like this to set up the test case:
```php
$country = (new CountryEntitiy)->assign([
    'customerTax' => new TaxFreeConfig(false),
    'companyTax' => new TaxFreeConfig(true),
    'vatIdPattern' => '...',
    'checkVatIdPattern' => true,
]);

$customer = (new CustomerEntity())->assign([
     'company' => 'ABC Compay',
     'vatIds' => [null],
]);

// create mock of sales channel context as you do now

// test detector as you do now
```

Reply:
and please add one empty line for each function, the CI is currently failing because of the missing empty lines.
Alternatively run `composer run ecs-fix -- src` to let our CS-Fixer run and fix code style issues automatically.

Reply:
Thanks for your feedback, i hope i got it right ✌️